<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>nanostringr Package • nanostringr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="nanostringr Package">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">nanostringr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/OVCARE/nanostringr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>nanostringr Package</h1>
            
            <h4 class="date">2019-03-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/OVCARE/nanostringr/blob/master/vignettes/Overview.Rmd"><code>vignettes/Overview.Rmd</code></a></small>
      <div class="hidden name"><code>Overview.Rmd</code></div>

    </div>

    
    
<p>The <code>nanostringr</code> R package is a companion R package to the manuscript:</p>
<p>A. Talhouk, S. Kommoss, R. Mackenzie, M. Cheung, S. Leung, D. Chiu, S. Kalloger, D. Huntsman, S. Chen, M. Intermaggio, J. Gronwald, F. Chan, S. Ramus, C. Steidl, D. Scott, M. Anglesio. (2016). Single-patient molecular testing with NanoString nCounter Data using a reference-based strategy for batch effect correction. <em>PLos ONE</em>.</p>
<p>This vignette provides a guide to reproduce the analyses in the paper and document the use of some of the functions.</p>
<div id="the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#the-data" class="anchor"></a>The Data</h1>
<p>Included in this package are several datasets that are described in detail in the manuscript and that are annotated in expQC, the annotation data frame for all of the experiments, run in different CodeSets, including:</p>
<ul>
<li>
<strong>Hodgkin Lymphoma Clinical Samples (HL)</strong> TOTAL 74 samples
<ul>
<li>HL1 n=32 Unique Samples</li>
<li>HL2 n=32 Replicates of HL1 samples</li>
<li>HL3 n=10 Replicates (subset of HL1 samples)</li>
</ul>
</li>
<li>
<strong>Ovarian Cancer Clinical Samples (OC)</strong> TOTAL 258 samples
<ul>
<li>OC1 n=129 Unique</li>
<li>OC2 n=129 Replicates of OC1 samples</li>
</ul>
</li>
<li>
<strong>Ovarian Cancer Cell Lines (OVCL)</strong> TOTAL 26
<ul>
<li>OC1 n=13 Unique</li>
<li>OC2 n=13 Replicates of OC1 samples</li>
</ul>
</li>
<li>
<strong>DNA Oligonucleotides for the HL CodeSet (HLO)</strong> TOTAL 68 HLO pool run at different concentrations
<ul>
<li>HL1 n=36<br>
</li>
<li>HL2 n=30<br>
</li>
<li>HL3 n=2</li>
</ul>
</li>
<li>
<strong>DNA Oligonucleotides for the OC CodeSet (OVO)</strong> TOTAL 135 OVO pool run at different concentrations
<ul>
<li>OC1 n=47
<ul>
<li>OC2 n=88</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div id="the-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#the-functions" class="anchor"></a>The Functions</h1>
<div id="nanostringqc" class="section level2">
<h2 class="hasAnchor">
<a href="#nanostringqc" class="anchor"></a>NanoStringQC</h2>
<p>The NanoStringQC function computes several QC metrics and appends them to the annotation matrix. Care should be taken to ensure that the data is in the proper format. The function returns different QC metric flags that can be used to filter out samples that fail QC.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(nanostringr)</a>
<a class="sourceLine" id="cb1-2" title="2">expOVD &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NanoStringQC.html">NanoStringQC</a></span>(ovd.r, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expQC, OVD <span class="op">==</span><span class="st"> "Yes"</span>))</a>
<a class="sourceLine" id="cb1-3" title="3">expOVO &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NanoStringQC.html">NanoStringQC</a></span>(ovo.r, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expQC, OVO <span class="op">==</span><span class="st"> "Yes"</span>))</a>
<a class="sourceLine" id="cb1-4" title="4">expOVCL &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NanoStringQC.html">NanoStringQC</a></span>(ovc.r, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expQC, OVCL <span class="op">==</span><span class="st"> "Yes"</span>))</a>
<a class="sourceLine" id="cb1-5" title="5">expHLD &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NanoStringQC.html">NanoStringQC</a></span>(hld.r, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expQC, HLD <span class="op">==</span><span class="st"> "Yes"</span>))</a>
<a class="sourceLine" id="cb1-6" title="6">expHLO &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NanoStringQC.html">NanoStringQC</a></span>(hlo.r, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expQC, HLO <span class="op">==</span><span class="st"> "Yes"</span>))</a>
<a class="sourceLine" id="cb1-7" title="7">expQC &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(expHLD, expOVD, expHLO, expOVO, expOVCL)</a>
<a class="sourceLine" id="cb1-8" title="8">expQC<span class="op">$</span>cohort &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"HLD"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(expHLD)), </a>
<a class="sourceLine" id="cb1-9" title="9">                <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"OVD"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(expOVD)), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"HLO"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(expHLO)),</a>
<a class="sourceLine" id="cb1-10" title="10">                <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"OVO"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(expOVO)), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"OVCL"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(expOVCL))))</a>
<a class="sourceLine" id="cb1-11" title="11">expQC &lt;-<span class="st"> </span>expQC <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-12" title="12"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">cohort =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(stringr<span class="op">::</span><span class="kw"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span>(cohort,</a>
<a class="sourceLine" id="cb1-13" title="13">                                                  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HLD"</span> =<span class="st"> "HL"</span>,</a>
<a class="sourceLine" id="cb1-14" title="14">                                                    <span class="st">"OVD"</span> =<span class="st"> "OC"</span>)),</a>
<a class="sourceLine" id="cb1-15" title="15">                         <span class="dt">levels =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HL"</span>, <span class="st">"OC"</span>, <span class="st">"OVCL"</span>, <span class="st">"HLO"</span>, <span class="st">"OVO"</span>)))</a></code></pre></div>
<div id="metrics-for-quality-assurance" class="section level3">
<h3 class="hasAnchor">
<a href="#metrics-for-quality-assurance" class="anchor"></a>Metrics for Quality Assurance</h3>
</div>
<div id="fields-of-view-fov" class="section level3">
<h3 class="hasAnchor">
<a href="#fields-of-view-fov" class="anchor"></a>Fields Of View (FOV)</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(perFOV <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">ylab =</span> <span class="st">"% FOV"</span>, <span class="dt">main =</span> <span class="st">"% FOV by Cohort"</span>, <span class="dt">data =</span> expQC, <span class="dt">pch =</span> <span class="dv">20</span>,</a>
<a class="sourceLine" id="cb2-2" title="2">        <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a>
<a class="sourceLine" id="cb2-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> <span class="dv">75</span>, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb2-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/grid">grid</a></span>(<span class="ot">NULL</span>, <span class="ot">NULL</span>, <span class="dt">lwd =</span> <span class="dv">1</span>)</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/perFOVPlot-1.png" alt="Samples that failed imaging QC based on percent fields of view (FOV) counted across cohorts." width="672"><p class="caption">
Samples that failed imaging QC based on percent fields of view (FOV) counted across cohorts.
</p>
</div>
</div>
<div id="positive-controls" class="section level3">
<h3 class="hasAnchor">
<a href="#positive-controls" class="anchor"></a>Positive Controls</h3>
<div id="linearity-of-positive-controls" class="section level4">
<h4 class="hasAnchor">
<a href="#linearity-of-positive-controls" class="anchor"></a>Linearity of Positive Controls</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(linPC <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">ylab =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/expression">expression</a></span>(R <span class="op">^</span><span class="st"> </span><span class="dv">2</span>), <span class="dt">main =</span> <span class="st">"Linearity of Positive Controls by Cohort"</span>,</a>
<a class="sourceLine" id="cb3-2" title="2">        <span class="dt">data =</span> expQC, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO), <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb3-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> <span class="fl">0.95</span>, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb3-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/grid">grid</a></span>(<span class="ot">NULL</span>, <span class="ot">NULL</span>, <span class="dt">lwd =</span> <span class="dv">1</span>)</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/linPCPlot-1.png" alt="Plot of $R^2$ of positive control probes from samples across all cohorts." width="672"><p class="caption">
Plot of <span class="math inline">\(R^2\)</span> of positive control probes from samples across all cohorts.
</p>
</div>
</div>
</div>
<div id="signal-to-noise-ratio-snr" class="section level3">
<h3 class="hasAnchor">
<a href="#signal-to-noise-ratio-snr" class="anchor"></a>Signal to Noise Ratio (SNR)</h3>
<div id="level-of-housekeeping-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#level-of-housekeeping-genes" class="anchor"></a>Level of Housekeeping Genes</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(averageHK <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">ylab =</span> <span class="st">"Average log HK expression"</span>,</a>
<a class="sourceLine" id="cb4-2" title="2">        <span class="dt">main =</span> <span class="st">"Average log expression of Housekeeping genes by Cohort"</span>, <span class="dt">data =</span> expQC, <span class="dt">pch =</span> <span class="dv">20</span>,</a>
<a class="sourceLine" id="cb4-3" title="3">        <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a>
<a class="sourceLine" id="cb4-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> <span class="dv">50</span>, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb4-5" title="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/grid">grid</a></span>(<span class="ot">NULL</span>, <span class="ot">NULL</span>, <span class="dt">lwd =</span> <span class="dv">1</span>)</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/averageHKPlot-1.png" alt="Average log expression of Housekeeping genes by Cohort." width="672"><p class="caption">
Average log expression of Housekeeping genes by Cohort.
</p>
</div>
</div>
<div id="limit-of-detection-lod" class="section level4">
<h4 class="hasAnchor">
<a href="#limit-of-detection-lod" class="anchor"></a>Limit of Detection (LOD)</h4>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(lod <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">ylab =</span> <span class="st">"LOD"</span>, <span class="dt">main =</span> <span class="st">"Limit of detection (LOD) by Cohort"</span>,</a>
<a class="sourceLine" id="cb5-2" title="2">        <span class="dt">data =</span> expQC, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a>
<a class="sourceLine" id="cb5-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> <span class="dv">50</span>, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb5-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/grid">grid</a></span>(<span class="ot">NULL</span>, <span class="ot">NULL</span>, <span class="dt">lwd =</span> <span class="dv">1</span>)</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/lodPlot-1.png" alt="Limit of detection by cohort." width="672"><p class="caption">
Limit of detection by cohort.
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(pergd <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">data =</span> expQC, <span class="dt">border =</span> <span class="st">"white"</span>,</a>
<a class="sourceLine" id="cb6-2" title="2">       <span class="dt">ylab =</span> <span class="st">"% Genes Detected"</span>, </a>
<a class="sourceLine" id="cb6-3" title="3">       <span class="dt">main =</span> <span class="st">"Percent of Genes Detected Above </span><span class="ch">\n</span><span class="st"> the Limit of Detection"</span>, </a>
<a class="sourceLine" id="cb6-4" title="4">       <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a>
<a class="sourceLine" id="cb6-5" title="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> <span class="dv">50</span>, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb6-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/grid">grid</a></span>(<span class="ot">NULL</span>, <span class="ot">NULL</span>, <span class="dt">lwd =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb6-7" title="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/stripchart">stripchart</a></span>(pergd <span class="op">~</span><span class="st"> </span>cohort, <span class="dt">data =</span> expQC, </a>
<a class="sourceLine" id="cb6-8" title="8">           <span class="dt">vertical =</span> <span class="ot">TRUE</span>, <span class="dt">method =</span> <span class="st">"jitter"</span>,  </a>
<a class="sourceLine" id="cb6-9" title="9">           <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">cex =</span> <span class="fl">0.4</span> , <span class="dt">col =</span> <span class="st">"#3A6EE3"</span>, </a>
<a class="sourceLine" id="cb6-10" title="10">           <span class="dt">add =</span> <span class="ot">TRUE</span>) </a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/pergdPlot-1.png" alt="Percent genes of total (excluding controls) detected above the limit of detection." width="672"><p class="caption">
Percent genes of total (excluding controls) detected above the limit of detection.
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">sn &lt;-<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb7-2" title="2">detect &lt;-<span class="st"> </span><span class="dv">60</span></a>
<a class="sourceLine" id="cb7-3" title="3"></a>
<a class="sourceLine" id="cb7-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(expOVD<span class="op">$</span>sn, expOVD<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVD, <span class="dt">xaxt =</span> <span class="st">"n"</span>, <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">100</span>), <span class="dt">xlim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/range">range</a></span>(expOVD<span class="op">$</span>sn),</a>
<a class="sourceLine" id="cb7-5" title="5">     <span class="dt">xlab =</span> <span class="st">"Signal to Noise Ratio"</span>, <span class="dt">ylab =</span> <span class="st">"% Genes Detected"</span>)</a>
<a class="sourceLine" id="cb7-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expOVO<span class="op">$</span>sn, expOVO<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVO)</a>
<a class="sourceLine" id="cb7-7" title="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expOVCL<span class="op">$</span>sn, expOVCL<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVCL)</a>
<a class="sourceLine" id="cb7-8" title="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expHLD<span class="op">$</span>sn, expHLD<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.HLD)</a>
<a class="sourceLine" id="cb7-9" title="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expHLO<span class="op">$</span>sn, expHLO<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.HLO)</a>
<a class="sourceLine" id="cb7-10" title="10"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/axis">axis</a></span>(<span class="dv">1</span>, <span class="dt">at =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">0</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">max</a></span>(expQC<span class="op">$</span>sn) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dv">300</span>))</a>
<a class="sourceLine" id="cb7-11" title="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">v =</span> sn, <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb7-12" title="12"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> detect, <span class="dt">lty =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb7-13" title="13"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/title">title</a></span>(<span class="st">"Signal to Noise vs </span><span class="ch">\n</span><span class="st"> Ratio of Genes Detected"</span>)</a>
<a class="sourceLine" id="cb7-14" title="14"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"bottomright"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HL"</span>, <span class="st">"OC"</span>, <span class="st">"OVCL"</span>, <span class="st">"HLO"</span>, <span class="st">"OVO"</span>), <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">bty =</span> <span class="st">'n'</span>,</a>
<a class="sourceLine" id="cb7-15" title="15">       <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/snPlot-1.png" alt="Signal to Noise versus % Gene Detected by cohort." width="672"><p class="caption">
Signal to Noise versus % Gene Detected by cohort.
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(expOVD<span class="op">$</span>sn, expOVD<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVD, <span class="dt">xaxt =</span> <span class="st">"n"</span>, <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">100</span>), <span class="dt">xlim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">6000</span>),</a>
<a class="sourceLine" id="cb8-2" title="2">     <span class="dt">xlab =</span> <span class="st">"Signal to Noise Ratio "</span>, <span class="dt">ylab =</span> <span class="st">"Ratio of Genes Detected"</span>)</a>
<a class="sourceLine" id="cb8-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expOVO<span class="op">$</span>sn, expOVO<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVO)</a>
<a class="sourceLine" id="cb8-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expOVCL<span class="op">$</span>sn, expOVCL<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.OVCL)</a>
<a class="sourceLine" id="cb8-5" title="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expHLD<span class="op">$</span>sn, expHLD<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.HLD)</a>
<a class="sourceLine" id="cb8-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(expHLO<span class="op">$</span>sn, expHLO<span class="op">$</span>pergd, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> COL.HLO)</a>
<a class="sourceLine" id="cb8-7" title="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/axis">axis</a></span>(<span class="dv">1</span>, <span class="dt">at =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">0</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">max</a></span>(expQC<span class="op">$</span>sn) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dv">300</span>))</a>
<a class="sourceLine" id="cb8-8" title="8"></a>
<a class="sourceLine" id="cb8-9" title="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">v =</span> sn, <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb8-10" title="10"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">h =</span> detect, <span class="dt">lty =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb8-11" title="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/title">title</a></span>(<span class="st">"Signal to Noise vs </span><span class="ch">\n</span><span class="st"> Ratio of Genes Detected (Zooming-in)"</span>)</a>
<a class="sourceLine" id="cb8-12" title="12"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"bottomright"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HL"</span>, <span class="st">"OC"</span>, <span class="st">"OVCL"</span>, <span class="st">"HLO"</span>, <span class="st">"OVO"</span>), <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">bty =</span> <span class="st">'n'</span>,</a>
<a class="sourceLine" id="cb8-13" title="13">       <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(COL.HLD, COL.OVD, COL.OVCL, COL.HLO, COL.OVO))</a></code></pre></div>
<div class="figure">
<img src="Overview_files/figure-html/snZoom-1.png" alt="Signal to Noise versus % Gene Detected by cohort zoomed in to the area of possible failures." width="672"><p class="caption">
Signal to Noise versus % Gene Detected by cohort zoomed in to the area of possible failures.
</p>
</div>
</div>
</div>
</div>
<div id="hknorm" class="section level2">
<h2 class="hasAnchor">
<a href="#hknorm" class="anchor"></a>HKnorm</h2>
<p>The HKnorm function does a simple normalization to log transformed (base 2) gene expression data data to housekeeping genes. This is done by subtracting the average log housekeeping gene expression level from the expression level of every gene in each sample.</p>
<p>As an example, here we check the QC metrics of the Hodgkin Lymphoma data and normalize it to housekeeping genes:</p>
<p>We can check to see if any samples failed QC metrics</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">expHLD0 &lt;-<span class="st"> </span>expHLD</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/any">any</a></span>(expHLD0<span class="op">$</span>QCFlag <span class="op">==</span><span class="st"> "Failed"</span>)</a></code></pre></div>
<p>[1] TRUE</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">expHLD0<span class="op">$</span>sampleID[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(expHLD0<span class="op">$</span>QCFlag <span class="op">==</span><span class="st"> "Failed"</span>)]</a></code></pre></div>
<p>[1] “HL1_18”</p>
<p>Since these are matched samples we must remove both pairs from the annotation data frame and from the gene expression data frame.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">expHLD &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(expHLD0, sampleID <span class="op">!=</span><span class="st"> "HL1_18"</span> <span class="op">&amp;</span><span class="st"> </span>sampleID <span class="op">!=</span><span class="st"> "HL2_18"</span>)</a>
<a class="sourceLine" id="cb11-2" title="2">hld &lt;-<span class="st"> </span>hld.r[, <span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(hld.r) <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HL1_18"</span>, <span class="st">"HL2_18"</span>)]</a></code></pre></div>
<p>We now normalize the resulting gene expression data</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1"><span class="co"># If data already log normalized</span></a>
<a class="sourceLine" id="cb12-2" title="2">hld.n &lt;-<span class="st"> </span><span class="kw"><a href="../reference/HKnorm.html">HKnorm</a></span>(hld, <span class="dt">is.logged =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb12-3" title="3"></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="co"># Otherwise, normalize to HK </span></a>
<a class="sourceLine" id="cb12-5" title="5">hld.n &lt;-<span class="st"> </span><span class="kw"><a href="../reference/HKnorm.html">HKnorm</a></span>(hld)</a>
<a class="sourceLine" id="cb12-6" title="6">hld1 &lt;-<span class="st"> </span>hld.n[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"HL1"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(hld.n))]</a>
<a class="sourceLine" id="cb12-7" title="7">exp.hld1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expHLD, geneRLF <span class="op">==</span><span class="st"> "HL1"</span>)</a>
<a class="sourceLine" id="cb12-8" title="8"></a>
<a class="sourceLine" id="cb12-9" title="9">hld2 &lt;-<span class="st"> </span>hld.n[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"HL2"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(hld.n))]</a>
<a class="sourceLine" id="cb12-10" title="10">exp.hld2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(expHLD, geneRLF <span class="op">==</span><span class="st"> "HL2"</span>)</a></code></pre></div>
</div>
<div id="refmethod" class="section level2">
<h2 class="hasAnchor">
<a href="#refmethod" class="anchor"></a>refMethod</h2>
<p>This function does batch adjustment using a reference-based strategy.</p>
<p>Below is how this would work for the HL data:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">r &lt;-<span class="st"> </span><span class="dv">3</span> <span class="co"># The number of references to use</span></a>
<a class="sourceLine" id="cb13-2" title="2">choice.refs &lt;-<span class="st"> </span>exp.hld1<span class="op">$</span>sampleID[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>((<span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(exp.hld1)[<span class="dv">1</span>]), r, <span class="dt">replace =</span> F)] <span class="co"># select reference samples randomly</span></a>
<a class="sourceLine" id="cb13-3" title="3">R1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld1[, choice.refs])</a>
<a class="sourceLine" id="cb13-4" title="4">R2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld2[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"HL2"</span>, <span class="kw">getNum</span>(choice.refs), <span class="dt">sep =</span> <span class="st">"_"</span>)])</a>
<a class="sourceLine" id="cb13-5" title="5">Y &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld2[, <span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(hld2) <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"HL2"</span>, <span class="kw">getNum</span>(choice.refs), <span class="dt">sep =</span> <span class="st">"_"</span>)])</a>
<a class="sourceLine" id="cb13-6" title="6">S2.r &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(<span class="kw"><a href="../reference/refMethod.html">refMethod</a></span>(Y, R1, R2)) <span class="co"># Data from CodeSet 2 now calibrated for CodeSet 1</span></a></code></pre></div>
<p>We can check the result by selecting a random gene and plotting the expression values from both CodeSets</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2016</span>)</a>
<a class="sourceLine" id="cb14-2" title="2">gene &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(<span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(hld1), <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb14-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/par">par</a></span>(<span class="dt">mfrow =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb14-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld1[gene, ]), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld2[gene, ]), <span class="dt">xlab =</span> <span class="st">"HL1"</span>, <span class="dt">ylab =</span> <span class="st">"HL2"</span>,</a>
<a class="sourceLine" id="cb14-5" title="5">     <span class="dt">main =</span> <span class="st">"No Correction"</span>)</a>
<a class="sourceLine" id="cb14-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dv">0</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb14-7" title="7"></a>
<a class="sourceLine" id="cb14-8" title="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(hld1[gene, <span class="op">!</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(hld1) <span class="op">%in%</span><span class="st"> </span>choice.refs)]), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(S2.r[gene, ]),</a>
<a class="sourceLine" id="cb14-9" title="9">     <span class="dt">xlab =</span> <span class="st">"HL1"</span>, <span class="dt">ylab =</span> <span class="st">"HL2"</span>, <span class="dt">main =</span> <span class="st">"Corrected"</span>)</a>
<a class="sourceLine" id="cb14-10" title="10"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dv">0</span>, <span class="dv">1</span>)</a></code></pre></div>
<p><img src="Overview_files/figure-html/plot_gene-1.png" width="672"></p>
<!-- # Downstream Analysis -->
<!-- In the previous analysis we compared the gene expression value of an individual gene. Below is the impact on a downstream analysis. We use the HL prognostic model as an example. For additional information on this model, the reader is referred to the [paper](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0153844). -->
<!-- ```{r downstream_analysis} -->
<!-- library(CHL26predictor) -->
<!-- # We select the genes that are used in the model -->
<!-- CHL26.HL1.exprs <- hld.n[rownames(hld.n) %in% CHL26.model.coef.df$geneName, -->
<!--                          grep("HL1", colnames(hld.n))] + log(1000, 2) -->
<!-- CHL26.HL2.exprs <- hld.n[rownames(hld.n) %in% CHL26.model.coef.df$geneName, -->
<!--                          grep("HL2", colnames(hld.n))] + log(1000, 2) -->
<!-- # This is the threshold -->
<!-- risk.thres <- 0.6235 -->
<!-- n <- dim(CHL26.HL1.exprs)[2] -->
<!-- # We compute the risk scores in each CodeSet -->
<!-- scores.df1 <- get_CHL26_scores(as.matrix(CHL26.HL1.exprs)) -->
<!-- scores.df2 <- get_CHL26_scores(as.matrix(CHL26.HL2.exprs)) -->
<!-- scores.risk.df1 <- scores.df1 %>% -->
<!--   mutate(riskClass = ifelse(score >= risk.thres, "High", "Low")) -->
<!-- scores.risk.df2 <- scores.df2 %>% -->
<!--   mutate(riskClass = ifelse(score >= risk.thres, "High", "Low")) -->
<!-- # We assess how many cases are misclassified -->
<!-- tabRisk <- table(scores.risk.df1$riskClass, scores.risk.df2$riskClass) -->
<!-- ind.mis <- which(scores.risk.df1$riskClass != scores.risk.df2$riskClass) -->
<!-- mis <- n - sum(diag(tabRisk)) -->
<!-- ``` -->
<!-- ```{r RawScores, message = FALSE, echo = FALSE, warning = FALSE} -->
<!-- CCplot(scores.risk.df1$score, scores.risk.df2$score, Ptype = "scatter", -->
<!--        xrange = range(scores.df1$score), yrange = range(scores.df2$score), -->
<!--        xlabel = "HL1", ylabel = "HL2", subtitle = "Scores without Batch Adjustment") -->
<!-- abline(h = risk.thres, col = 2, lty = 1) -->
<!-- text(0.7, 0.58, paste("Misclassified:", mis), col = "red") -->
<!-- points(scores.risk.df1$score[ind.mis], scores.risk.df2$score[ind.mis], col = "red") -->
<!-- ``` -->
<!-- ```{r BootStrapped, message = FALSE, echo = FALSE, warning = FALSE, cache = TRUE} -->
<!-- set.seed(40) -->
<!-- r <- 3  # Number of reference samples -->
<!-- nB <- 1500  # of bootstrap samples -->
<!-- nth <- 1000 -->
<!-- misCount <- rep(0, nB) -->
<!-- ind.mis.count <- rep(0, nB) -->
<!-- Cmetrix <- matrix(0, nrow = nB, ncol = 3) -->
<!-- for (i in 1:nB) { -->
<!--   choice.refs <- exp.hld1$sampleID[sample((1:dim(exp.hld1)[1]), r, replace = F)]  # select reference samples randomly -->
<!--   DSR1 <- t(hld1[, choice.refs] + log2(1000)) -->
<!--   DSR2 <- t(hld2[, paste("HL2", getNum(choice.refs), sep = "_")] + log2(1000)) -->
<!--   DSY <- t(hld2[, !colnames(hld2) %in% paste("HL2", getNum(choice.refs), sep = "_")] + log2(1000)) -->
<!--   DSS2.r <- t(refMethod(DSY, DSR1, DSR2)) -->
<!--   CHL26.HL2.r.SS.exprs <- DSS2.r[rownames(DSS2.r) %in% CHL26.model.coef.df$geneName, ] -->
<!--   scores.ss.df2.r <- get_CHL26_scores(as.matrix(CHL26.HL2.r.SS.exprs)) -->
<!--   scores.risk.ss.df2.r <- scores.ss.df2.r %>% -->
<!--     mutate(riskClass = ifelse(score >= risk.thres, "High", "Low")) -->
<!--   ndx2.r <- substring(scores.df1$sampleID, 5) %in% substring(scores.ss.df2.r$sampleID, 5) -->
<!--   misCount[i] <- (n - r) - sum(diag(table(scores.risk.df1$riskClass[ndx2.r], -->
<!--                                           scores.risk.ss.df2.r$riskClass))) -->
<!--   ind.mis.t <- scores.risk.ss.df2.r$sampleID[which(scores.risk.df1$riskClass[ndx2.r] != scores.risk.ss.df2.r$riskClass)] -->
<!--   ind.mis.count[i] <- (sum(ind.mis.t %in% c("HL2_30", "HL2_32")) == misCount[i]) -->
<!--   Cmetrix[i, ] <- CCplot(scores.df1$score[ndx2.r], scores.risk.ss.df2.r$score, metrics = TRUE) -->
<!--   if (i %% nth == 0) { -->
<!--     CCplot(scores.df1$score[ndx2.r], scores.risk.ss.df2.r$score, Ptype = "scatter", -->
<!--            xrange = range(scores.df1$score), yrange = range(scores.df2$score), -->
<!--            xlabel = "HL1", ylabel = "HL2") -->
<!--     abline(h = risk.thres, col = 2, lty = 1) -->
<!--     text(0.7, 0.58, paste("Misclassified:", misCount[i]), col = "red") -->
<!--   } -->
<!-- } -->
<!-- TotalMisCount <- table(misCount) / nB -->
<!-- Accuracy <- Cmetrix[, 2] -->
<!-- knitr::kable(data.frame(table(Accuracy))) -->
<!-- if (length(TotalMisCount) > 3) { -->
<!--   print("more than 3 misclassifications observed") -->
<!-- } -->
<!-- ``` -->
<!-- Another visualization to assess concordance correlation is the Bland-Altman plot or the MA plot: -->
<!-- ```{r MAplot, echo=FALSE} -->
<!-- Metrics <- CCplot(scores.risk.df1$score, scores.risk.df2$score, -->
<!--                   Ptype = "MAplot", -->
<!--                   xrange = range(scores.df1$score), -->
<!--                   yrange = range(scores.df2$score), -->
<!--                   xlabel = "HL1", ylabel = "HL2", metrics = TRUE) -->
<!-- knitr::kable(data.frame(Metrics)) -->
<!-- ``` -->
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-data">The Data</a></li>
      <li>
<a href="#the-functions">The Functions</a><ul class="nav nav-pills nav-stacked">
<li><a href="#nanostringqc">NanoStringQC</a></li>
      <li><a href="#hknorm">HKnorm</a></li>
      <li><a href="#refmethod">refMethod</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Derek Chiu, Aline Talhouk, Samuel Leung.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
